ibis/backends/sql/compiler.py

```diff
@@ -954,6 +954,8 @@ def visit_VarianceStandardDevCovariance(self, op, *, how, where, **kw):
     )
 
     def visit_SimpleCase(self, op, *, base=None, cases, results, default):
+        if not cases:
+            return default
```

Comment:
`cases` should never be empty.

For example, why would someone ever write this:

```python
t.x.cases(else_="foo")
```

Reply:
This feels similar to whether or not we should allow empty Table schemas and empty structs.

No one would ever write this literally, but I personally ran into this with some code like `branches = [(case, result) for case, result in branches if not is_bad(result)]`

Supporting empty branches is rarely necessary and able to be solvable externally, but semantically it is well-defined, consistent, and unsurprising: the whole thing evals to `else_`.

I lean towards supporting it, but since it can be solved externally we can limit the scope.

Also, probably I can adjust this to catch the empty case at the `ibis.cases()` level and just return `else_`, and then avoid even creating an ops.SimpleCase

Reply:
Please drop this change so we can merge this PR.

When you introduce a change that adds new behavior like this it's going to make the review take much longer.

The fewer behavioral changes and decisions that are made in a PR, the faster we can merge it. It's much easier to debate something like this when it's not buried inside of another giant changeset.